### PR TITLE
HTBHF-2675 Changed select-address to store address data in session.stepData instead of res.locals

### PR DESCRIPTION
**Pull request by YetAnotherMatt**
Also validating that an address has been selected

### CODE REVIEW

File: src/web/routes/application/steps/address/select-address/select-address.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ const { transformAddress } = require('./adapters')
 const { addressContentSummary } = require('../content-summary')
 const { requestBody } = require('../request-body')
 const { stateMachine, actions } = require('../../../flow-control')
+const { validate } = require('./validate')
+const { setAdditionalDataForStep } = require('../../../flow-control/session-accessors')
```
Comment (gethyn1) [inline src/web/routes/application/steps/address/select-address/select-address.js]: This should be exported from `flow-control` - see line 5

File: src/web/routes/application/steps/address/select-address/select-address.js [hunk 2]
```diff
@@ -33,14 +35,24 @@ const resetAddressState = (req) => {
   }
 }
 
-const behaviourForGet = (config, journey) => (req, res, next) => {
+const getAddressDataFromSession = (req, addressId) => {
+  const addresses = req.session.postcodeLookupResults.map(buildAddressOption(addressId))
```
Comment (gethyn1) [inline src/web/routes/application/steps/address/select-address/select-address.js]: You could inline this on line 41

Comment (YetAnotherMatt) [reply src/web/routes/application/steps/address/select-address/select-address.js]: it's also referenced on line 42, so I don't think I can?

Comment (gethyn1) [reply src/web/routes/application/steps/address/select-address/select-address.js]: oh, i didn't see that

File: src/web/routes/application/steps/address/select-address/select-address.js (hunk 1 above)
Comment (YetAnotherMatt) [reply src/web/routes/application/steps/address/select-address/select-address.js]: done, thanks

**Pull request merged**